transformer_lens/hook_points.py

```diff
@@ -45,63 +71,77 @@ def __init__(self):
+            def full_forward_hook(
+                module: torch.nn.Module, module_input: Any, module_output: torch.Tensor
+            ):
                 return hook(module_output, hook=self)
 
-            full_hook.__name__ = (
+            full_forward_hook.__name__ = (
                 hook.__repr__()
-            )  # annotate the `full_hook` with the string representation of the `hook` function
+            )  # annotate the `full_forward_hook` with the string representation of the `hook` function
 
-            handle = self.register_forward_hook(full_hook)
+            handle = self.register_forward_hook(full_forward_hook)
             handle = LensHandle(handle, is_permanent, level)
 
             if prepend:
-                # we could just pass this as an argument in PyTorch 2.0, but for now we manually do this...
-                self._forward_hooks.move_to_end(handle.hook.id, last=False)
-                self.fwd_hooks.insert(0, handle)
+                # we could just pass this as an argument in PyTorch 2.0, but for now we manually do
+                # this...
 
+                self._forward_hooks.move_to_end(handle.hook.id, last=False)  # type: ignore[attr-defined]
+                self.fwd_hooks.insert(0, handle)
             else:
                 self.fwd_hooks.append(handle)
-
         elif dir == "bwd":
             # For a backwards hook, module_output is a tuple of (grad,) - I don't know why.
-
-            def full_hook(module, module_input, module_output):
+            def full_backward_hook(
+                module: torch.nn.Module, module_input: _grad_t, module_output: _grad_t
+            ):
                 return hook(module_output[0], hook=self)
 
-            full_hook.__name__ = (
+            full_backward_hook.__name__ = (
                 hook.__repr__()
-            )  # annotate the `full_hook` with the string representation of the `hook` function
+            )  # annotate the `full_backward_hook` with the string representation of the `hook` function
 
-            handle = self.register_full_backward_hook(full_hook)
+            handle = self.register_full_backward_hook(full_backward_hook)
             handle = LensHandle(handle, is_permanent, level)
 
             if prepend:
                 # we could just pass this as an argument in PyTorch 2.0, but for now we manually do this...
-                self._backward_hooks.move_to_end(handle.hook.id, last=False)
+                self._backward_hooks.move_to_end(handle.hook.id, last=False)  # type: ignore[attr-defined]
                 self.bwd_hooks.insert(0, handle)
             else:
                 self.bwd_hooks.append(handle)
         else:
             raise ValueError(f"Invalid direction {dir}")
 
-    def remove_hooks(self, dir="fwd", including_permanent=False, level=None) -> None:
+    def remove_hooks(
+        self,
+        dir: str = "fwd",
+        including_permanent: bool = False,
+        level: Union[int, None] = None,
```

Comment:
```suggestion
        level: Optional[int] = None,
```

Reply:
nit: This is the same, but more standard. Will help us find-and-replace all optional types in the codebase once Optional and Union both get removed (in Python 3.13 I think).